additions to your library

xesscorp opened the pull request:
I created an alternate symbol library with the pin types set to reflect their actual functions (in the original library, all the pins are set as 'input'). I generated the new symbol library from the CSV file using KiPart so my symbols don't have the graphic elements found in the originals. That's why I put them in a separate symbol library.

Commit:
Added a new symbol library (esp32-xess.lib) that was generated using KiPart on the esp32-xess.csv file.
The esp32-xess library has:

    * Pins of the ESP32 chip and modules are assigned types that reflect their
      functions (e.g., 'bidirectional' instead of 'input'). This helps detect
      errors when running the ERC in EESCHEMA.

    * There is a schematic symbol for the ESP32 chip with the pins arranged
      as they are in the QFN48 package.

Commit:
Added image of ESP32_QUAD schematic symbol to README.md.

adamjvr commented:
Good call on finding the pins not having the type set, I recently pushed and update where I edited my symbols to have the correct type set according to actual function.

xesscorp commented:
A couple of other things I've noticed:
1. The pins of the QFN48 footprint are arranged in a clockwise manner, but the Espressif pinout for the ESP32 shows the pins in a counter-clockwise arrangement.
2. The upper-left pad of the ESP3212 footprint is given the pin number 21, but I think it should be pin number 1.

adamjvr commented:
Would you be willing to save your modified symbols in the existing library rather than there own file ? I'm more than willing to include additional symbols since everyone likes to have their schematics look one way or the other, but I'd strongly prefer it was all in the same library file etc.

xesscorp commented:
I was avoiding doing that because I didn't want to stomp on your existing library. We would need a way to avoid symbol name collisions.

I'm currently working on a combined library for the ESP8266 and ESP32. When that's done (which is taking longer than expected), I'll let you know and then you can decide which (if any) symbols to include in your library.

adamjvr commented:
Ok, I understand now. Let's see what your new library looks like. I'm coming over from using Altium for years so tend to design my symbols a lot like that, and they tend to be untraditional. I'll be checking back here for your new lib, and we can see what we can combine or perhaps include as a secondary library.

adamjvr commented:
Any updates on your library ?

Commit:
Added espressif-xess.lib with many symbols for ESP8266 and ESP32 chips and modules.

xesscorp commented:
I added a new library with both the ESP8266 and ESP32 chips and many modules in it. I also created the .dcm file with the descriptions, keywords and document links.

There are two symbols for both the ESP8266 and ESP32: one that shows a quad-sided chip with short pins names, and another that is single-sided with fully elaborated pin names.

There are several potential problems:
1. The labeling of GPIO4 and GPIO5 is inconsistent with several of the modules, most notable the ESP-12. I took my best guess.
2. The numbering of the pins on many modules is unspecified, so I oriented the module with the PCB antenna at the top and assigned numbers starting at the upper-left corner of the module and proceeded counter-clockwise. One exception to this is the ESP-201 module that seems to be oriented with the antenna at the bottom.

Take a look and modify as needed.

adamjvr commented:
Almost forgot about this. Heres what I was thinking, I'd accept your pull request but then import your footprints into the current library with the _L at the end of the name L to denote "literal" since your symbols correlate directly to how the parts themselves are pinned. Now that being said it appears that the ESP3212 has been canceled and none have been shipped out rather its been replaced with the ESP-32S and only those have been shipped out. It appears to have the same dimensions and specifications as the ESP-32-WROOM module, seems just like AI-thinker's own respin of the reference module. Let me know what you think of my suggestion and how you wanna proceed given the new change.

xesscorp commented:
On 10/29/2016 7:09 PM, Adam Vadala-Roth wrote:

> Almost forgot about this. Heres what I was thinking, I'd accept your 
> pull request but then import your footprints into the current library 
> with the _L at the end of the name L to denote "literal" since your 
> symbols correlate directly to how the parts themselves are pinned. Now 
> that being said it appears that the ESP3212 has been canceled and none 
> have been shipped out rather its been replaced with the ESP-32S and 
> only those have been shipped out. It appears to have the same 
> dimensions and specifications as the ESP-32-WROOM module, seems just 
> like AI-thinker's own respin of the reference module. Let me know what 
> you think of my suggestion and how you wanna proceed given the new change.

Your proposal sounds good. It's really your repository, so anything is fine.

I agree that the ESP3212 symbol should be removed.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> https://github.com/adamjvr/ESP32-kiCAD-Footprints/pull/3#issuecomment-257121376, 
> or mute the thread 
> https://github.com/notifications/unsubscribe-auth/AAznG5uxixh4Lkg6r9nlJkLPrFEoQsaVks5q49IVgaJpZM4J_MBM.

## 

---

Dave Vandenbout / XESS Corp.
2608 Sweetgum Drive
Apex NC 27539-8851
USA
/devb@xess.com/

## /www.xess.com/

adamjvr commented:
Alright so it says the branch has some conflicting files, if you wanna fix that, I'll accept the pull request and then add the symbols with the _L to the main library and credit you in the readme.

Commit:
Merge branch 'master' of https://github.com/adamjvr/ESP32-kiCAD-Footprints

# Conflicts:
#	README.md

In addition, all XESS-related Esspressif parts (ESP8266 & ESP32) are now in espressif-xess.* files. esp32-xess.* files were removed.

xesscorp commented:
I removeded the conflicts on the README.

I removed the ESP3212 from the library and replaced it with the ESP32S. I noticed that the ESP32S has the same pinout as the ESP32-WROOM. Let me know if you think that's not correct.

I also removed the esp32-xess.\* files since the ESP32 parts are now integrated with the ESP8266 parts in the espressif-xess.\* files.
